### ISSUE TITLE
Feature request - toggle consent during desktop connect

### ISSUE THREAD

**Issue by cgb** [Ylianst/MeshCentral#1153]
Generally I have the connection toolbar & notify enabled for a group of desktops, but don't require consent for conveneint remote access.

What do you think of the idea of permitting right-click on the Desktop connect button, similar to how admin shell / powershell is triggered, and provide the user with a 'connect+prompt for consent' option for those times when you want the user to opt-in to the remote support session?

**Comment by Ylianst** [comment]
Yes, this is possible, I would have to work on it.

**Comment by Ylianst** [comment]
Also, a quick note to mention that as of a few days ago, you can set the user consent settings for devices individually. Not the same as this request, but fairly quick to change.

![x](https://user-images.githubusercontent.com/1319013/79212759-e080f600-7dfc-11ea-8083-f077cb96b067.png)

**Comment by cgb** [comment]
Thanks Yilan, I was browsing through your recent commits and did see the per-device consent options, which is a fantastic addition.  It gave me the idea for toggling consent on a per-device basis... :)

**Comment by Ylianst** [comment]
Just published MeshCentral v0.5.1-u with a bunch of web application improvements including right-click on the remote desktop connect button to ask for user consent. Let me know if this works.

![MC2-WebAppImprovements](https://user-images.githubusercontent.com/1319013/79299044-d06a2480-7e97-11ea-8e33-349fc7be827a.png)

**Comment by cgb** [comment]
Tested successfully, thank you so much for this, unbelievelable turn around between feature request & implementation!

**Comment by iviamandi** [comment]
Observation: if you have already set user with ask consent enabled, right click on Desktop Connect shows ask consent, but logs without user consent.
A user could log without consent, although he has not this right. Not ok.

**Comment by amenolo** [comment]
I can confirm the misbehavior and BTW could worth add similar action to Connect button under Terminal and Files tabs.

**Comment by Ylianst** [comment]
OH!!! That is really bad. Working on this now!

**Comment by Ylianst** [comment]
Published MeshCentral v0.5.1-v with a fix for this. The problem was a single character in the code:

```
before: ws.httprequest.consent += obj.consent;
after: ws.httprequest.consent |= obj.consent;
```

This version also has improved night mode.

**Comment by amenolo** [comment]
BTW along with user consent could be useful enable `connection toolbar` too so to have a temporary complete "consent mode", isn't it?

**Comment by Ylianst** [comment]
MeshCentral v0.5.1-w now has two options on the right-click connect.

![x](https://user-images.githubusercontent.com/1319013/79410516-82b3f180-7f55-11ea-9d22-59a38263e829.png)

**Comment by Ylianst** [comment]
I think I can close this one since it's been released. Please don't hesitate to open a new issue if you find any problems. Thanks.
